extensions/wikia/ArticleExporter/ArticleExporterHooks.class.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+use Wikia\Rabbit\ConnectionBase;
+
+/**
+ * @author Lore team
+ *
+ * Send plaintext versions of articles to RabbitMQ as they are edited on Fandom
+ */
+
+class ArticleExporterHooks {
+    protected static $rabbitConnection;
+
+    public static function onArticleSaveComplete( WikiPage $page ) {
+        global $wgCityId;
+        global $wgArticleExporterExchange;
+
+        $message = [
+            'wikiId' => $wgCityId,
+            'pageIds' => [ strval($page->getId()) ],
+        ];
+
+        self::getRabbitConnection()->publish( $wgArticleExporterExchange['routing'], $message );
+
+        return true;
+    }
+
+    protected static function getRabbitConnection() {
+        global $wgArticleExporterExchange;
+
+        if ( !isset( self::$rabbitConnection ) ) {
+            self::$rabbitConnection = new ConnectionBase( $wgArticleExporterExchange );
+        }
```

Comment:
Can this be injected? I thought we had some form of dependency injection in MW...

Reply:
We do, but I'm not sure anyone uses it :(

With the unified platform I could see that being one bit that goes away, maybe.

Reply:
Yeah this is generally the way we've been doing things (at least from all of the extensions I've seen).